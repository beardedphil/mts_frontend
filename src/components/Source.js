import React from 'react';

const Source = ({ source }) => {
    return (
        <div className="source">
            <p>ID: { source.id }</p>
            <p>Source: { source.source }</p>
            <p>Proper Source: { source.proper_source }</p>
        </div>
    );
};

export default Source;
