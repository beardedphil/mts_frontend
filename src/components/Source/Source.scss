@import '../../styles/settings';

.source {
  border-radius: 8px;
  box-shadow:
          -12px -12px 12px 0 $neu-upper,
          12px 12px 12px 0 $neu-lower;
  margin: 20px 30px;
  padding: 20px;
  width: 320px;
}

.source .source-form {
  align-items: center;
  display: flex;
}

.source img {
  border-radius: 8px;
  margin-left: 20px;
  width: 50px;
}

.source p {
  padding-left: 90px;
}

.selected {
  background: $highlight-gray;
}
