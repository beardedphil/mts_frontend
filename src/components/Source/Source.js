import React, { useState, useEffect } from 'react';
import { useTrustedSources } from '../contexts/TrustedSourceContext';
import './Source.scss';

const Source = ({ source }) => {
    const [trusted, setTrusted] = useState(false);
    const [trustedSources, setTrustedSources] = useTrustedSources();

    useEffect(() => {
        if (trustedSources.includes(source.source)) {
            setTrusted(true);
        }
    }, []);

    const updateTrustedSources = (e) => {
        setTrusted(e.target.checked);
        const index = trustedSources.indexOf(source.source);
        // Moving from unchecked to checked
        if (e.target.checked) {
            if (index === -1) {
                const updatedSources = [...trustedSources, source.source]
                setTrustedSources(updatedSources);
                const json = JSON.stringify(updatedSources);
                localStorage.setItem('sources', json);
            }
        // Moving from checked to unchecked
        } else {
            if (index > -1) {
                const updatedSources = trustedSources.filter(item => item !== source.source);
                setTrustedSources(updatedSources);
                const json = JSON.stringify(updatedSources);
                localStorage.setItem('sources', json);
            }
        }
    }

    return (
        <div className={`source ${ trusted && 'selected' }`}>
            <form className="source-form">
                <input type="checkbox" checked={trusted} onChange={updateTrustedSources} />
                <img src={`./images/${ source.source }.png`} alt={source.source}/>
                <p>{ source.proper_source }</p>
            </form>
        </div>
    );
};

export default Source;
