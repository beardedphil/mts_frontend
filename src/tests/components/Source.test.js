import React from 'react';
import { shallow } from 'enzyme';
import Source from '../../components/Source';
import sources from '../fixtures/sources';

test('should render Source correctly', () => {
    const wrapper = shallow(<Source source={ sources[1] } />);
    expect(wrapper).toMatchSnapshot();
});
