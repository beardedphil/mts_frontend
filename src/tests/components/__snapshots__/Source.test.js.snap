// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Source correctly 1`] = `
<div
  className="source"
>
  <p>
    ID: 
    2
  </p>
  <p>
    Source: 
    slate
  </p>
  <p>
    Proper Source: 
    Slate
  </p>
</div>
`;
